src/crimson/mgr/client.cc

```diff
@@ -152,9 +152,13 @@ seastar::future<> Client::handle_mgr_conf(crimson::net::ConnectionRef,
 void Client::report()
 {
   gate.dispatch_in_background(__func__, *this, [this] {
-    assert(conn);
-    auto pg_stats = with_stats.get_stats();
-    return conn->send(std::move(pg_stats));
+    if (conn) {
+      auto pg_stats = with_stats.get_stats();
+      return conn->send(std::move(pg_stats));
+    } else {
+      logger().warn("report: no conn available; raport skipped");
+      return seastar::now();
+    }
```

Comment:
nit

```suggestion
    if (!conn) {
      logger().warn("report: no conn available; raport skipped");
      return seastar::now();
    }
    auto pg_stats = with_stats.get_stats();
    return conn->send(std::move(pg_stats));
```

early return to highlight the normal (happy) path.